### PR TITLE
Update to Kubernetes v1.28.3

### PR DESCRIPTION
**Pull request by bertinatto**
CC  @soltysh

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go [hunk 1]
```diff
@@ -294,14 +294,8 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	StorageVersionHash: {Default: true, PreRelease: featuregate.Beta},
 
-<<<<<<< HEAD
-	UnauthenticatedHTTP2DOSMitigation: {Default: true, PreRelease: featuregate.Beta},
-
-||||||| 89a4ea3e1e4
-=======
```
Comment (bertinatto) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: In upstream, this was disabled by https://github.com/kubernetes/kubernetes/pull/121196/commits/0f33a62f9768cf7120d93b74015f5bde3df9a477.

Also, switching the gate here wouldn't be enough. If this gate is to be enabled, it also needs to be switched in `/pkg/features/kube_features.go`

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by soltysh** [comment]
/override ci/prow/verify-commits 
this never passes on k8s bump

**Pull request merged**